src/EthPool.sol

```diff
@@ -77,7 +77,11 @@ contract EthPool is Pool {
             );
     }
 
-    function _transfer(address to, uint128 amt) internal override {
-        if (amt != 0) payable(to).transfer(amt);
+    function _transfer(address userAddr, int128 amt) internal override {
+        // Take into account the amount already transferred into the pool
+        amt += int128(uint128(msg.value));
+        if (amt == 0) return;
+        require(amt > 0, "Sending a negative ether amount");
```

Comment:
```suggestion
        require(amt > 0, "not-enough-eth-sent");
```

Reply:
Might be easier to understand for the user why it failed

Reply:
eth transfer logic seems to work

Reply:
This is more of a sanity check and a message to the developer who breaks the transfer logic. It should never fail, the user can't break it, but can't fix it either, sending more eth won't help :D